### PR TITLE
Python 3 compat

### PR DESCRIPTION
**Pull request by brandond**

### CODE REVIEW

File: aws_adfs/login.py [hunk 1]
```diff
@@ -172,31 +176,11 @@ def _authenticate(config, username=None, password=None):
...
-                    saml2attribute.iter('{urn:oasis:names:tc:SAML:2.0:assertion}AttributeValue')),
-                filter(
-                    lambda saml2attribute: saml2attribute.get('Name') == 'https://aws.amazon.com/SAML/Attributes/Role',
-                    saml.iter('{urn:oasis:names:tc:SAML:2.0:assertion}Attribute')
-                ),
-            )
-        )
-    )
+    # Find all roles offered by the assertion
+    aws_roles = [element.text.split(',') for element in saml.findall('.//{*}Attribute[@Name="https://aws.amazon.com/SAML/Attributes/Role"]/{*}AttributeValue')]
```
Comment (venth) [inline aws_adfs/login.py]: Now it's much more readable 👍

File: aws_adfs/login.py [hunk 2]
```diff
@@ -151,7 +155,7 @@ def _authenticate(config, username=None, password=None):
     del password
 
     # Decode the response
-    html = ET.fromstring(response.text.decode('utf8'), ET.HTMLParser())
+    html = ET.fromstring(response.text, ET.HTMLParser())
```
Comment (venth) [inline aws_adfs/login.py]: I'll create a docker container to verify that one. Wonder if it'll be sufficient to confirm that the bug exists and them was fixed.

Comment (brandond) [inline aws_adfs/login.py]: It didn't work at all as-is in Python 3; I am not sure about the charset issue that was reported with Py2.

Comment (venth) [inline aws_adfs/login.py]: @Brandond would you like to exclude this change from pull request. I'll add Travis and couple of test to verify that tool works with different versions of Python. I'll try to find out how to reproduce issue added by @dzinek.

**Comment by venth** [comment]
@brandond, I've refactored a bit login.py. It isn't ready yet, but it could give "a taste" about the direction. Would you like to review it and share your opinion with me? The branch name: feature/travis_and_test_verification

File: aws_adfs/login.py (hunk 2 above)
Comment (brandond) [inline aws_adfs/login.py]: Leaving the decode('utf8') in breaks Python 3 support.
```
  File "/home/ec2-user/python34-env/lib/python3.4/site-packages/aws_adfs/login.py", line 72, in login
    principal_roles, assertion = _authenticate(config)
  File "/home/ec2-user/python34-env/lib/python3.4/site-packages/aws_adfs/login.py", line 158, in _authenticate
    html = ET.fromstring(response.text.decode('utf8'), ET.HTMLParser())
AttributeError: 'str' object has no attribute 'decode'
```

Removing it works fine for me with both Python 2.7 and 3.4; are you seeing something else?

Comment (venth) [inline aws_adfs/login.py]: No, you right. At least for os with UTF-8 set as a base encoding. I think, it can be safely removed.

**Pull request merged**